### PR TITLE
Adds Guidelines for Framework Modules Contributions

### LINKED ISSUE #2648: Adds Guidelines for Framework Modules Contributions
#### Expected Behavior
Have guidelines for framework modules contributions.

#### Actual Behavior
Framework modules have some patterns that must be observed by contributors, but there is no guidelines.

#### Potential Solution
Add guidelines in CONTRIBUTING.md.

#### Checklist
- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)

### PR DESCRIPTION
**Pull request by AlexCatarino**
#### Related Issue
Closes #2648 

#### Motivation and Context
Include guidelines for framework modules contributions to help users to commit new modules knowing the pattern they should follow.

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Non-functional change (xml comments/documentation/etc)

#### Checklist:
- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description` or `feature-<issue#>-<description>`

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -20,6 +20,14 @@ As a point of consistency, we use soft tabs of four spaces to ensure files rende
 
 All pull requests must be accompanied by units tests. If it is a new feature, the tests should highlight expected use cases as well as edge cases, if applicable. If it is a bugfix, there should be tests that expose the bug in question.
 
+## Guidelines for Framework Modules Contributions
+
+Contributions of [Algorithm Framework](https://www.quantconnect.com/docs/algorithm-framework/overview) Modules needs to follow certain extra patterns, since QuantConnect users can use them in their algorithms.
+
+Generally modules should do one focused, specific role well. For example, combining risk control logic with [notifications](https://www.quantconnect.com/docs/live-trading/notifications) or placing orders outside execution models violates the general programming rule 'separation of concerns'. Keep each module doing one specific task and if you want to consider additional functionality add event handlers that users can bind to from their Algorithm instance.
```
Comment (mchandschuh) [inline CONTRIBUTING.md]: Great comment on event handlers! I think that's an exceptional pattern to use for enabled custom models to give the algorithm direct access to certain events that might be detected and be worthy of logging/notifications/etc 👍

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -20,6 +20,14 @@ As a point of consistency, we use soft tabs of four spaces to ensure files rende
...
+
+By default production code should be silent unless there is a fatal exception. Because of this, [logging or debugging](https://www.quantconnect.com/docs/algorithm-reference/logging-and-debug) is not allowed inside LEAN framework modules. Additional [charting](https://www.quantconnect.com/docs/algorithm-reference/charting) inside the module consumes the resources and should not be included in a module as well.
```
Comment (mchandschuh) [inline CONTRIBUTING.md]: Regarding logging/debugging -- we do have `QCAlgorithmFramework.DebugEnabled` -- perhaps allowing logging/debugging when `DebugEnabled == true` makes sense?

File: CONTRIBUTING.md (hunk 1 above)
Comment (AlexCatarino) [reply CONTRIBUTING.md]: It was @jaredbroad's idea. I think we could add a couple of examples as soon as possible,
